devfile-support/samples/spring-boot-http-booster-devfile.yaml

```diff
@@ -0,0 +1,99 @@
+schemaVersion: 2.0.0
+metadata:
+  name: spring-boot-http-booster
+projects:
+  - name: spring-boot-http-booster
+    git:
+      location: https://github.com/snowdrop/spring-boot-http-booster
+      branch: master
+components:
+  - chePlugin:
+      registryEntry:
+        id: redhat/java8/latest
+  - chePlugin:
+      registryEntry:
+        id: redhat/dependency-analytics/latest
+  - container:
+      name: maven
+      image: registry.redhat.io/codeready-workspaces/stacks-java-rhel8:2.1
+      mountSources: true
+      memoryLimit: 768Mi
+      env:
+        - name: JAVA_OPTS
+          value: >-
+            -XX:MaxRAMPercentage=50.0 -XX:+UseParallelGC -XX:MinHeapFreeRatio=10
+            -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4
+            -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true
+            -Xms20m -Djava.security.egd=file:/dev/./urandom -Duser.home=/home/jboss
+        - name: MAVEN_OPTS
+          value: $(JAVA_OPTS)
+      endpoints:
+        - name: 8080-tcp
+          targetPort: 8080
+          configuration:
+              public: true
+      volumeMounts:
+        - name: m2
+          path: /home/jboss/.m2
+  - volume:
+      name: m2
+      size: 2G
+commands:
+  - exec:
+      id: build 
+      component: maven
+      commandLine: mvn -Duser.home=${HOME} -DskipTests clean install
+      workingDir: '${CHE_PROJECTS_ROOT}/spring-boot-http-booster'
+      env:
+        - name: MAVEN_OPTS
+          value: "-Xmx200m"
```

Comment:
Does it just override the value from container configuration?
If so - is it correct? container seems to have more options that are just reset and it may lead to build failure.
Maybe it makes sense to use another example env var here...

Reply:
This is not directly an override of the container component `env` field.
But it still overrides this env variable in the container for this command, since it simply defines env variables for this particular command.
Just as if the command was:
`export MAVEN_OPTS='-Xmx200m' ; mvn -Duser.home=${HOME} -DskipTests clean install`